pkg/frontend/transport/handler.go

```diff
@@ -347,6 +347,9 @@ func (f *Handler) reportQueryStats(
 			"results_cache_hit_bytes", details.ResultsCacheHitBytes,
 			"results_cache_miss_bytes", details.ResultsCacheMissBytes,
 		)
+		if details.ResultsCacheNoStore {
+			logMessage = append(logMessage, "results_cache_no_store", true)
```

Comment:
I suggest to always add the log field, not just when it's true. I think it makes logs easier to parse.

Reply:
Tested locally :

header not present:

`curl -G   http://localhost:8001/prometheus/api/v1/query_range --data-urlencode 'query=up{x="y"}' --data-urlencode 'start=1719993614' --data-urlencode 'end=1719995414' --data-urlencode 'step=1m'`

```
mimir-1-1        | ts=2024-07-03T08:32:10.867257807Z caller=handler.go:383 level=info user=anonymous traceID=6ee0dc648a604a48 msg="query stats" component=query-frontend method=GET path=/prometheus/api/v1/query_range route_name=prometheus_api_v1_query_range user_agent=curl/7.68.0 status_code=200 response_time=7.371747ms response_size_bytes=63 query_wall_time_seconds=0.005392547 fetched_series_count=0 fetched_chunk_bytes=0 fetched_chunks_count=0 fetched_index_bytes=0 sharded_queries=0 split_queries=1 estimated_series_count=0 queue_time_seconds=5.35e-05 param_step=60000 param_query="up{x=\"y\"}" param_start=2024-07-03T08:00:14Z param_end=2024-07-03T08:30:14Z length=35m0s time_since_min_time=36m56.859773194s time_since_max_time=1m56.859773194s results_cache_hit_bytes=0 results_cache_miss_bytes=81 
results_cache_no_store=false status=success
```

header present
`curl -G -H "Cache-Control: no-store"   http://localhost:8001/prometheus/api/v1/query_range --data-urlencode 'query=up{x="y"}' --data-urlencode 'start=1719993614' --data-urlencode 'end=1719995414' --data-urlencode 'step=1m'`


```
mimir-1-1        | ts=2024-07-03T08:32:43.265653417Z caller=handler.go:383 level=info user=anonymous traceID=51702283e2e53281 msg="query stats" component=query-frontend method=GET path=/prometheus/api/v1/query_range route_name=prometheus_api_v1_query_range user_agent=curl/7.68.0 status_code=200 response_time=5.211026ms response_size_bytes=63 query_wall_time_seconds=0.003876335 fetched_series_count=0 fetched_chunk_bytes=0 fetched_chunks_count=0 fetched_index_bytes=0 sharded_queries=0 split_queries=1 estimated_series_count=0 queue_time_seconds=5.0635e-05 param_end=2024-07-03T08:30:14Z param_step=60000 param_query="up{x=\"y\"}" param_start=2024-07-03T08:00:14Z length=35m0s time_since_min_time=37m29.260357532s time_since_max_time=2m29.260357532s results_cache_hit_bytes=0 results_cache_miss_bytes=81 
results_cache_no_store=true status=success
```